### PR TITLE
EVG-7624 add git tag version aliases to project

### PR DESCRIPTION
**Pull request by ablack12**
image

### CODE REVIEW

File: public/static/js/projects.js [hunk 1]
```diff
@@ -563,17 +568,29 @@ mciModule.controller('ProjectCtrl', function ($scope, $window, $http, $location,
...
-
   };
 
+  $scope.addGitTagAlias = function() {
+    if (!$scope.validGitTagVersionDefinition($scope.git_tag_alias)) {
+      $scope.invalidGitTagAliasMessage = "An alias must have a git tag alias, variant regex, and exactly one of task regex or tag";
+      return;
+    }
+    item = Object.assign({}, $scope.git_tag_alias);
+    item["alias"] = "__git_tag";
```
Comment (john-m-liu) [inline public/static/js/projects.js:579]: I would only access properties like this if it's a variable. item.alias is easier to read

File: public/static/js/projects.js [hunk 2]
```diff
@@ -781,14 +806,33 @@ mciModule.controller('ProjectCtrl', function ($scope, $window, $http, $location,
     return true;
   }
 
+  $scope.patchDefinitionPopulated = function(alias) {
+    return alias && (Boolean(alias.variant) || Boolean(alias.task) || !_.isEmpty(alias.variant_tags) || !_.isEmpty(alias.tags));
```
Comment (john-m-liu) [inline public/static/js/projects.js:810]: You don't need to convert these to bools, it'll happen automatically

File: public/static/js/projects.js [hunk 3]
```diff
@@ -781,14 +806,33 @@ mciModule.controller('ProjectCtrl', function ($scope, $window, $http, $location,
...
+  }
+
+  $scope.aliasRemotePathPopulated = function(alias) {
+    return Boolean(alias.remote_path);
```
Comment (john-m-liu) [inline public/static/js/projects.js:814]: A more succinct way to convert to boolean is by doing !!, or just ! if you want to invert it

Comment (ablack12) [reply public/static/js/projects.js:814]: I personally prefer Boolean to !! because it's more readable.

**Pull request merged**